esp/services/ws_dfu/ws_dfuXRefService.cpp

```diff
@@ -279,33 +251,23 @@ bool CWsDfuXRefEx::onDFUXRefFoundFiles(IEspContext &context, IEspDFUXRefFoundFil
     {
         context.ensureFeatureAccess(FEATURE_URL, SecAccess_Read, ECLWATCH_DFU_XREF_ACCESS_DENIED, "WsDfuXRef::DFUXRefFoundFiles: Permission denied.");
 
-        StringBuffer username;
-        context.getUserID(username);
-
-        if (!req.getCluster() || !*req.getCluster())
-            throw MakeStringExceptionDirect(ECLWATCH_INVALID_INPUT, "Cluster not defined.");
-
-        Owned<IXRefNode> xRefNode = XRefNodeManager->getXRefNode(req.getCluster());
-        if (xRefNode.get() == 0)
-            throw MakeStringExceptionDirect(ECLWATCH_CANNOT_FIND_IXREFFILESNODE, "XRefNode not found.");
+        Owned<IXRefNode> xRefNode = getXRefNodeByCluster(req.getCluster());
+        Owned<IXRefFilesNode> foundFiles = xRefNode->getFoundFiles();
+        if (!foundFiles)
+            return true;
 
         StringBuffer buf;
-        Owned<IXRefFilesNode> _found = xRefNode->getFoundFiles();
-        if (_found)
-        {
-            _found->Serialize(buf);
-            if (!buf.isEmpty())
-            {
-                ESPSerializationFormat fmt = context.getResponseFormat();
-                if (fmt == ESPSerializationJSON)
-                    dfuXrefXMLToJSON(buf);
-            }
-        }
-        resp.setDFUXRefFoundFilesQueryResult(buf.str());
+        foundFiles->Serialize(buf);
+        if (buf.isEmpty())
+            return true;
+
+        if (context.getResponseFormat() == ESPSerializationJSON)
+            dfuXrefXMLToJSON(buf);
```

Comment:
How hard would it be to update foundFiles->Serialize(buf); to support either xml or json?  these types of converters are less efficient, but also treat all json fields as strings which loses it's typing benefits.

Reply:
I will take a look when I have time.

Reply:
@afishbeck looks like that the foundFiles->Serialize(buf) only reads the xml. If I trace the code correctly, the xml is read from dali. I believe that the xml is created when DFUXRef is built.

Reply:
@wangkx ok, makes sense, just thought it was worth considering.